src/db_parser.jl

```diff
@@ -63,545 +353,1075 @@ function map_prime_mover(prime_mover::String)
+    for rec in DBInterface.execute(db, QUERIES[:zones])
+        component_attr = get(attributes, rec.id, Dict{String, Any}())
+        a = Area(;
+            name=rec.name,
+            load_response=component_attr["load_response"],
+            peak_active_power=component_attr["peak_active_power"],
+            peak_reactive_power=component_attr["peak_reactive_power"],
+        )
+        if haskey(component_attr, "uuid")
+            IS.set_uuid!(IS.get_internal(a), Base.UUID(component_attr["uuid"]))
         end
-    else
-        type = "Forecast"
-        # Create a dictionary to store daily values
-        daily_values = Dict{String, Vector{Float64}}()
-
-        # Iterate over each row in the DataFrame
-        for row in eachrow(df)
-            # Extract the date part of the timestamp
-            date_part = join(split(row["timestamp"], "-")[1:3], "-")
-
-            # Initialize the daily values array if not already present
-            if !haskey(daily_values, date_part)
-                daily_values[date_part] = Float64[]
-            end
+        PSY.add_component!(p.base_system, a)
+    end
+    for rec in DBInterface.execute(db, QUERIES[:balancing_topologies])
+        component_attr = get(attributes, rec.id, Dict{String, Any}())
+        area_name = first(DBInterface.execute(db, QUERIES[:zone], [rec.area])).name
+        b = PSY.ACBus(;
+            name=rec.name,
+            number=rec.id,
+            bustype=PSY.get_enum_value(PSY.ACBusTypes, component_attr["bustype"]),
+            angle=component_attr["angle"],
+            magnitude=component_attr["magnitude"],
+            area=get_component(Area, p.base_system, area_name),
+            voltage_limits=nothing,
+            base_voltage=component_attr["base_voltage"],
+            load_zone=nothing,
+        )
+        if haskey(component_attr, "uuid")
+            IS.set_uuid!(IS.get_internal(b), Base.UUID(component_attr["uuid"]))
+        end
+        PSY.add_component!(p.base_system, b)
+    end
+end
 
-            # Append the value to the daily values array
-            push!(daily_values[date_part], row["value"])
+function add_aggregate_lines!(
+    p::Portfolio,
+    attributes::Dict{Int64, Dict{String, Any}},
+    db::SQLite.DB,
+)
+    for rec in DBInterface.execute(db, QUERIES[:aggregate_lines])
+        component_attr = get(attributes, rec.id, Dict{String, Any}())
+        t = AggregateTransportTechnology{ACBranch}(;
+            name=string(rec.rowid),
+            id=rec.rowid,
+            available=true,
+            base_power=100.0,
+            power_systems_type=string(ACBranch),
+            financial_data=DEFAULT_FINANCIAL_DATA,
+            start_region=collect(
+                IS.get_components(
+                    x -> get_id(x) == parse(Int64, rec.area_from),
+                    RegionTopology,
+                    p.data,
+                ),
+            )[1],
```

Comment:
Is there a guarantee that this always returns an array of size 1? Would it be better to store a mapping ahead of time?